transformer_engine/pytorch/triton_kernels/cast.py

```
@@ -72,7 +72,8 @@ def te_quantize_triton(
             _setup_conditional_transpose_storage(out)
         else:
             out = quantizer.make_empty(input_tensor.shape, dtype=fake_tensor_type)
-            _setup_conditional_transpose_storage(out)
+            if isinstance(out, Float8TensorBase):
```

What happens if out isn't Float8TensorBase? Do we need to throw an error there?

---

If it isn't Float8TensorBase, it'll be a mxfp8TensorBase in which case we don't need to do transpose. Hence no need to throw any error.